cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -513,7 +513,8 @@ impl<T: Config> Pallet<T> {
 
 		let max_message_size =
 			T::ChannelInfo::get_channel_max(recipient).ok_or(MessageSendError::NoChannel)?;
-		if data.len() > max_message_size {
+		let format_size = format.encoded_size();
+		if data.len() + format_size > max_message_size {
```

Comment:
What about `saturating_add`?

Reply:
`checked_add` even and reject in the error case.